common-util/src/main/java/org/sunbird/common/validator/location/BaseLocationRequestValidator.java

```diff
@@ -69,4 +81,133 @@ public void validateSearchLocationRequest(Request req) {
           ResponseCode.CLIENT_ERROR.getResponseCode());
     }
   }
+
+  /**
+   * This method will validate the list of location code whether its valid or not. If valid will
+   * return the locationId List.
+   *
+   * @param codeList list of location code
+   * @param actorRef
+   * @return List<String> list of locationId
+   */
+  public static List<String> validateLocationCode(List<String> codeList, Object actorRef) {
+    Set<String> locationIds = null;
+    List<Map<String, Object>> locationList = null;
+    locationList = locationClient.getLocationsByCodes(codeList, actorRef);
+    List<String> locationIdList = new ArrayList<>();
+    if (!CollectionUtils.isEmpty(locationList)) {
+      locationIds = BaseLocationRequestValidator.validateLocationHierarchy(locationList, actorRef);
+    } else {
+      throw new ProjectCommonException(
+          ResponseCode.invalidParameterValue.getErrorCode(),
+          ProjectUtil.formatMessage(
+              ResponseCode.invalidParameterValue.getErrorMessage(),
+              codeList,
+              JsonKey.LOCATION_CODE),
+          ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+    locationIdList.addAll(locationIds);
+    return locationIdList;
+  }
+
+  /**
+   * This method will validate the location hierarchy and return the locationIds list
+   *
+   * @param locationList list of location
+   * @param actorRef actor reference
+   * @return list of locationId
+   */
+  public static Set<String> validateLocationHierarchy(
+      List<Map<String, Object>> locationList, Object actorRef) {
+    Set<String> locnIds = new HashSet<>();
+    Map<String, Object> responseLocnMapList = new HashMap<>();
+    for (Map<String, Object> location : locationList) {
+      Map<String, Object> parentLocnList = getParentLocations(location, actorRef);
+      addLocations(responseLocnMapList, parentLocnList);
+    }
+    locnIds.addAll(responseLocnMapList.keySet());
+    return locnIds;
+  }
+
+  private static void addLocations(
+      Map<String, Object> responseLocnMapList, Map<String, Object> parentLocnList) {
+    if (MapUtils.isEmpty(responseLocnMapList)) {
+      responseLocnMapList.putAll(parentLocnList);
+    } else {
+      checkLocationAlreadyExist(responseLocnMapList, parentLocnList);
+    }
+  }
+
+  private static void checkLocationAlreadyExist(
+      Map<String, Object> responseLocnMapList, Map<String, Object> parentLocnList) {
+    for (Object obj : parentLocnList.values()) {
+      Map<String, Object> currentLocation = (Map<String, Object>) obj;
+      String type = (String) currentLocation.get(JsonKey.TYPE);
+      for (Object object : responseLocnMapList.values()) {
```

Comment:
Check if we can use functional programming, map class method or lambda expressions to simplify if there is another element with same type and different id.